src/platform/common/application/commands.ts

```diff
@@ -180,7 +180,11 @@ export interface ICommandNameArgumentTypeMapping extends ICommandNameWithoutArgu
     [DSCommands.ShowDataViewer]: [IShowDataViewerFromVariablePanel];
     [DSCommands.RefreshDataViewer]: [];
     [DSCommands.ClearSavedJupyterUris]: [];
-    [DSCommands.SelectJupyterURI]: [boolean | undefined, Uri | 'toolbar' | 'nativeNotebookStatusBar' | undefined];
+    [DSCommands.SelectJupyterURI]: [
+        boolean | undefined,
+        Uri | 'toolbar' | 'nativeNotebookStatusBar' | undefined,
+        NotebookDocument | undefined
```

Comment:
@rchiodo should we change this to execCommand('selectJupyter...', {uri: Uri, document: NotebookDocument})`
I kinda feel thats a bit better than the current overloads.
E.g. the first argument `local= true` is not used an having everyone to pass a bogus value of `false` makes it unnecessary & feels like a poorly constructed API (I understand its historical)
& feel going with an object is better long term.

Reply:
I.e. we can keep backwards compatibility & also have the new Object syntax for new code going forward.

Reply:
The problem is there is a 3rd party using this command. If we change the arguments it will break them.

Reply:
> The problem is there is a 3rd party using this command. If we change the arguments it will break them.

We could make it backwards compatible.
I.e. if we get an object as the first argument, then we know its the new style. Because the first argument today is a `boolean|undefined`.

Reply:
Yes, its another overload, but going forward I kind of like the object approach, makes it very flexible.

Reply:
approving PR as this is definitely not a blocker, but think its worth exploring

Reply:
Seems weird to me